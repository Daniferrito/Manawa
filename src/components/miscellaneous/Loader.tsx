import React from 'react';

const Loader = () => <h1>Waiting...</h1>;

export default Loader;
