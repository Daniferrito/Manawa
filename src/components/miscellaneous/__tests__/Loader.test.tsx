import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Loader from '../Loader';
import renderer from 'react-test-renderer';

afterEach(cleanup);

test('renders without crashing', () => {
  const { baseElement } = render(<Loader />);
  expect(baseElement).toBeDefined();
});

test('matches the snapshots', () => {
  const component = renderer.create(<Loader />);
  expect(component).toMatchSnapshot();
});

test('sets the text to the right thing', () => {
  const component = render(<Loader />);
  expect(component.queryByText(/Waiting.../)).toBeTruthy();
});
