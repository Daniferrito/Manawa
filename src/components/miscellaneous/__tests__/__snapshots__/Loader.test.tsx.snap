// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshots 1`] = `
<h1>
  Waiting...
</h1>
`;
